pkg/sql/stmtdiagnostics/statement_diagnostics.go

```diff
@@ -411,37 +417,35 @@ func (r *Registry) CancelRequest(ctx context.Context, requestID int64) error {
 }
 
 // IsExecLatencyConditionMet returns true if the completed request's execution
-// latency satisfies the request's condition. If false is returned, it inlines
-// the logic of RemoveOngoing.
+// latency satisfies the request's condition. The request is automatically
+// removed from the registry (unless the continuous collection is enabled and
+// the request hasn't expired).
```

Comment:
This inlining of logic (before this PR) reads confusingly. Can it be pulled out into the one caller instead?